presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -0,0 +1,1321 @@
+    }
+
+    private void setPartitionColumnStatistics(
+            MetastoreContext metastoreContext,
+            String databaseName,
+            String tableName,
+            String partitionName,
+            Map<String, HiveType> columns,
+            Map<String, HiveColumnStatistics> columnStatistics,
+            OptionalLong rowCount)
+    {
+        List<ColumnStatisticsObj> metastoreColumnStatistics = columnStatistics.entrySet().stream()
+                .filter(entry -> columns.containsKey(entry.getKey()))
+                .map(entry -> createMetastoreColumnStatistics(entry.getKey(), columns.get(entry.getKey()), entry.getValue(), rowCount))
+                .collect(toImmutableList());
+        if (!metastoreColumnStatistics.isEmpty()) {
+            setPartitionColumnStatistics(metastoreContext, databaseName, tableName, partitionName, metastoreColumnStatistics);
+        }
+    }
+
+    private void setPartitionColumnStatistics(MetastoreContext metastoreContext, String databaseName, String tableName, String partitionName, List<ColumnStatisticsObj> statistics)
+    {
+        try {
+            retry()
+                    .stopOn(NoSuchObjectException.class, InvalidObjectException.class, MetaException.class, InvalidInputException.class)
+                    .stopOnIllegalExceptions()
+                    .run("setPartitionColumnStatistics", stats.getUpdatePartitionColumnStatistics().wrap(() ->
+                            getMetastoreClientThenCall(metastoreContext, client -> {
+                                client.setPartitionColumnStatistics(databaseName, tableName, partitionName, statistics);
+                                return null;
+                            })));
+        }
+        catch (NoSuchObjectException e) {
+            throw new TableNotFoundException(new SchemaTableName(databaseName, tableName));
+        }
+        catch (TException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+        catch (Exception e) {
+            throw propagate(e);
+        }
+    }
+
+    private void deletePartitionColumnStatistics(MetastoreContext metastoreContext, String databaseName, String tableName, String partitionName, String columnName)
+    {
+        try {
+            retry()
+                    .stopOn(NoSuchObjectException.class, InvalidObjectException.class, MetaException.class, InvalidInputException.class)
+                    .stopOnIllegalExceptions()
+                    .run("deletePartitionColumnStatistics", stats.getUpdatePartitionColumnStatistics().wrap(() ->
+                            getMetastoreClientThenCall(metastoreContext, client -> {
+                                client.deletePartitionColumnStatistics(databaseName, tableName, partitionName, columnName);
+                                return null;
+                            })));
+        }
+        catch (NoSuchObjectException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)